blackduck/HubRestApi.py

```diff
@@ -966,6 +966,22 @@ def delete_project_version_codelocations(self, version):
             logging.info("Deleting code location at: {}".format(code_location_url))
             self.execute_delete(code_location_url)
 
+    def delete_empty_projects(self):
+        #get all projects with no mapped code locations and delete them all
+        projects_response = self.get_projects()
+        if projects_response['totalCount'] > 0:
+            projects = projects_response['items']
+        for p in projects:
+            p_empty = True
+            versions = self.get_project_versions(p)
+            #if versions['totalCount'] == 1:
+            for v in versions:
+                codelocations = self.get_version_codelocations(versions['items'][0])
+                if codelocations['totalCount'] != 0:
+                    p_empty = False
```

Comment:
Consider breaking (from the for-loop) at this point? i.e. if we have found at least one version that has codelocations mapped to it, why bother looking for more?